### PR TITLE
fix: reclaim the space the vertical scrollbars occupy when invisible in `UITable` elements.

### PR DESCRIPTION
**Pull request by s4my**

### CODE REVIEW

File: luigi2.h [hunk 1]
```diff
@@ -3024,7 +3024,11 @@ int _UITableMessage(UIElement *element, UIMessage message, int di, void *dp) {
 	if (message == UI_MSG_PAINT) {
 		UIPainter *painter = (UIPainter *) dp;
 		UIRectangle bounds = element->bounds;
-		bounds.r -= UI_SIZE_SCROLL_BAR * element->window->scale;
+
+		if (table->vScroll->page < table->vScroll->maximum) {
```
Comment (nakst) [inline luigi2.h:3028]: This could be simplified to be `bounds.r = table->vScroll->bounds.l;` I think?

File: luigi2.h [hunk 2]
```diff
@@ -3099,9 +3103,15 @@ int _UITableMessage(UIElement *element, UIMessage message, int di, void *dp) {
...
 	} else if (message == UI_MSG_LAYOUT) {
 		UIRectangle scrollBarBounds = element->bounds;
-		scrollBarBounds.l = scrollBarBounds.r - UI_SIZE_SCROLL_BAR * element->window->scale;
 		table->vScroll->maximum = table->itemCount * UI_SIZE_TABLE_ROW * element->window->scale;
 		table->vScroll->page = UI_RECT_HEIGHT(element->bounds) - UI_SIZE_TABLE_HEADER * table->e.window->scale;
+
+		if (table->vScroll->page < table->vScroll->maximum) {
+			scrollBarBounds.l = scrollBarBounds.r - UI_SIZE_SCROLL_BAR * element->window->scale;
+		} else {
+			scrollBarBounds.l = scrollBarBounds.r * element->window->scale;
```
Comment (nakst) [inline luigi2.h:3112]: The `* element->window->scale` is not needed here.

File: luigi2.h (hunk 1 above)
Comment (s4my) [reply luigi2.h:3028]: `UIScrollBar` has no member called bounds ?

File: luigi2.h (hunk 2 above)
Comment (s4my) [reply luigi2.h:3112]: No it is needed, i test it. Set the `scale=2` for e.g. in the `gf2_config.ini` and you'll see.

File: luigi2.h (hunk 1 above)
Comment (nakst) [reply luigi2.h:3028]: Sorry, `table->vScroll->e.bounds.l`

Comment (s4my) [reply luigi2.h:3028]: yeah that won't work either, it breaks clicking and dragging the vertical scroll bars.

**Comment by s4my** [comment]
@nakst if you could merge this PR today that would be nice, i need it for testing a feature am working on.

**Comment by s4my** [comment]
Thanks. #133 should be the same.

**Comment by nakst** [comment]
I'm holding off on the changes to `UICode` so that I can test it properly. I want to check that it interacts with horizontal scroll bars correctly.

I've gone and made some of the simplifying changes I commented here in commit [820abcc](https://github.com/nakst/gf/commit/820abccec9e668d223704afafe2b0d708b169380). If there is a problem with them, please let me know specifically what the issues are, so I can revert my changes if necessary.

**Comment by s4my** [comment]
> I'm holding off on the changes to `UICode` so that I can test it properly. I want to check that it interacts with horizontal scroll bars correctly.

In that case you should start with #132

> I've gone and made some of the simplifying changes I commented here in commit [820abcc](https://github.com/nakst/gf/commit/820abccec9e668d223704afafe2b0d708b169380). If there is a problem with them, please let me know specifically what the issues are, so I can revert my changes if necessary.

I don't seem to find any problems with this, if i find anything in the futur, i'll tell you.

**Pull request merged**